python/pyarrow/tests/test_convert_builtin.py

```diff
@@ -1894,18 +1937,30 @@ def test_dictionary_from_strings():
     assert a.dictionary.equals(expected_dictionary)
 
 
-def _has_unique_field_names(ty):
-    if isinstance(ty, pa.StructType):
-        field_names = [field.name for field in ty]
-        return len(set(field_names)) == len(field_names)
-    else:
-        return True
+@pytest.mark.parametrize('unit', ['s', 'ms', 'us', 'ns'])
+def test_duration_array_roundtrip_corner_cases(unit):
+    # corner case discovered by hypothesis
+    ty = pa.duration(unit)
+    arr = pa.array([-2147483000], type=ty)
+    restored = pa.array(arr.to_pylist(), type=ty)
+    assert arr.equals(restored)
+
+
+@pytest.mark.pandas
+def test_duration_array_roundtrip_nanosecond_resolution_pandas_timedelta():
+    # corner case discovered by hypothesis
```

Comment:
Can you also mention the actual bug it is testing? (i.e. preserving the nanoseconds on conversion from a list of Timedelta objects, I think?)

Reply:
Added comment.